joomla/joomla-cms
# Improve remind an PW request / reset messages

Pull Request for Issue https://github.com/joomla/joomla-cms/pull/15307

### Summary of Changes

- Make sure when we are not in debug mode there is a "general" message displayed.
- On PW reset atempts for SU add the docs page & send the admin a mail explaining why he cant do this and how to do it.

This has been reported to the JSST by @brianteeman but has been decided to do it in the public tracker as this is a longstanding and public issue anyway. cc @joomla/security 

### Testing Instructions

- install a site whith this patch applyed
- create an non-superuser
- try to reset the non-superuser
- all should work correctly but with a different message
- try to reset the superuser
- there is now the same message but in the background a new mail is send to the super user telling him that he cant do what he wanted to do but how to do it.
- enable joomla debug mode (and also make sure error reporting is active.
- try to reset the super admin again
- you now get the `old` error message but with the docs linked.

### Actual result BEFORE applying this Pull Request

Different messages for different cases that could be used to do `user emuration`

### Expected result AFTER applying this Pull Request

One general message used for all cases
In the special case of the SU there is a mail in the background to inform him about the steps he has to take.

### Documentation Changes Required

None.
